File: artwork-service/src/main/java/com/gallery/artworkservice/infrastructure/ArtworkImageEntity.java
package com.gallery.artworkservice.infrastructure;

import jakarta.persistence.*;

@Entity
@Table(name = "ARTWORK_IMAGE")
public class ArtworkImageEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "ImageUrl", nullable = false)
    private String imageUrl;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ArtworkId", nullable = false)
    private ArtworkEntity artwork;

    public ArtworkImageEntity() {}

    public ArtworkImageEntity(String imageUrl, ArtworkEntity artwork) {
        this.imageUrl = imageUrl;
        this.artwork = artwork;
    }

    // Getters and setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }

    public ArtworkEntity getArtwork() { return artwork; }
    public void setArtwork(ArtworkEntity artwork) { this.artwork = artwork; }
}

File: app/ArtworkService/Infrastructure/Entities/ArtworkImageEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using ArtworkService.Domain;

namespace ArtworkService.Infrastructure.Entities
{
    [Table("ARTWORK_IMAGE", Schema = "dbo")]
    public class ArtworkImageEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("Id")]
        public int Id { get; set; }

        [Column("ArtworkId")]
        [Required]
        public int ArtworkId { get; set; }

        [Column("ImageUrl")]
        [Required]
        public string ImageUrl { get; set; }

        public ArtworkImageEntity()
        {
            Id = 0;
            ArtworkId = 0;
            ImageUrl = "";
        }

        public ArtworkImageEntity(ArtworkImage image)
        {
            Id = image.Id;
            ArtworkId = image.ArtworkId;
            ImageUrl = image.ImageUrl;
        }

        public ArtworkImage ToArtworkImage()
        {
            return new ArtworkImage(Id, ArtworkId, ImageUrl);
        }
    }
}